kythe/cxx/indexer/textproto/recordio_textparser.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KYTHE_CXX_INDEXER_TEXTPROTO_RECORDIO_TEXTPARSER_H_
+#define KYTHE_CXX_INDEXER_TEXTPROTO_RECORDIO_TEXTPARSER_H_
+
+#include "absl/types/optional.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/dynamic_message.h"
+#include "google/protobuf/text_format.h"
+
+namespace kythe {
+namespace lang_textproto {
+
+// Parses recordio formatted content separated by record delimiter. Callback
+// is invoked for each chunk that is parsed into proto message along with the
+// line offset of where that chunk begins in the file.
+//
+// Empty lines can be record delimiter and they can also separate two chunks
```

Comment:
Really, that's a case we need to handle?

Reply:
I've seen in google3... and when I just pass a chunk with just comment to TextFormat::Parser, it fails.